src/component/Item.js

```diff
@@ -24,15 +25,36 @@ class Item extends Component {
       data: undefined
     };
   }
+  /**
+   *
+   *
+   * @memberof Item
+   */
+  redirectToNotFound() {
+    this.props.history.replace({ pathname: ROUTES.NOT_FOUND });
+  }
 
+  /**
+   *
+   *
+   * @memberof Item
+   * @param {object} data
+   */
+  checkIfContentIsStory(data) {
+    if (data.type !== 'story') {
+      this.redirectToNotFound();
+    } else {
+      this.setState({
+        type: data.type,
+        data: data
+      });
+    }
+  }
   componentDidMount = async () => {
     const itemId = this.props.match.params.id;
     const data = await fetchItem(itemId);
 
-    this.setState({
-      type: data.type,
-      data: data
-    });
+    data === null ? this.gotoNotFound() : this.checkIfContentIsStory(data);
```

Comment:
Here you are violating the single responsibility principle. You are invoking a function `checkIfContentIsStory` that not justs checks if it's a story but also sets the state in some cases. Can we do something like:

```
if (data && data.type === 'story') {
     this.setState({
        type: data.type,
        data: data
      });
} else {
   this.gotoNotFound();
}
```